Core/SqlPreparation/SqlPreparationExpressionVisitor.cs

```diff
@@ -266,9 +267,8 @@ protected override Expression VisitConstant (ConstantExpression expression)
     {
       ArgumentUtility.CheckNotNull ("expression", expression);
 
-      var collection = expression.Value as ICollection;
-      if (collection != null)
-        return new SqlCollectionExpression (expression.Type, collection.Cast<object>().Select (Expression.Constant).Cast<Expression>());
+      if (ReflectionUtility.TryGetCollection (expression.Value, out var collection))
```

Comment:
Please check if this solved
https://re-motion.atlassian.net/browse/RMLNQSQL-24
(there's no PR, but RMLNQSQL-24 might be a subset of our current task)

Reply:
It does not implement the feature, because we decided to only support `ICollection`,` ICollection<T>`, and `IReadOnlyCollection<T>`, not `IEnumerable` or `IEnumerable<T>`, because a lazily evaluated enumerable can result in unexpected behavior.